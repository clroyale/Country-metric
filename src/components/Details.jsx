import React from 'react';
import { useSelector } from 'react-redux';
import { useLocation } from 'react-router-dom';
import styles from '../styles/Details.module.css';

function Details() {
  const { countries } = useSelector((state) => state.countries);
  const location = useLocation();
  const Id = location.state;
  const filteredCountry = countries.filter((country) => country.id === Id);

  return (
    <div className={styles.main_container}>
      <div className={styles.coat_of_arms_name_container}>
        <div className={styles.coat_of_arms_container}>
          <img className={styles.coat_of_arms} src={filteredCountry[0].coatOfArms.png || filteredCountry[0].flags.png} alt="" />
        </div>
        <div className={styles.country_name_container}>
          <h2 className={styles.country_name}>{filteredCountry[0].name.common}</h2>
        </div>
      </div>
      <div>
        <h2 className={styles.break_down_text}>Information break down</h2>
      </div>
      <ul className={styles.information_section}>
        <li className={styles.info_list}>
          <span className={styles.info_title}>Official name:</span>
          {filteredCountry[0].name.official}
        </li>
        <li className={styles.info_list}>
          <span className={styles.info_title}>Capital:</span>
          {filteredCountry[0].capital}
        </li>
        <li className={styles.info_list}>
          <span className={styles.info_title}>Continent:</span>
          {filteredCountry[0].region}
        </li>
        <li className={styles.info_list}>
          <span className={styles.info_title}>Subregion:</span>
          {filteredCountry[0].subregion}
        </li>
        <li className={styles.info_list}>
          <span className={styles.info_title}>Population:</span>
          {filteredCountry[0].population}
        </li>
        <li className={styles.info_list}>
          <span className={styles.info_title}>Area:</span>
          {`${filteredCountry[0].area} kmÂ²`}
        </li>
      </ul>

    </div>
  );
}

export default Details;
