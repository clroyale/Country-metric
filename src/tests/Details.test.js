import React from 'react';
import renderer from 'react-test-renderer';
import { useSelector } from 'react-redux';
import { useLocation } from 'react-router-dom';
import Details from '../components/Details';

jest.mock('react-redux', () => ({
  useSelector: jest.fn(),

}));

jest.mock('react-router-dom', () => ({
  useLocation: jest.fn(),
}));

describe('Details', () => {
  const mockData = [
    {
      name: { common: 'country one' }, population: 1234, flags: { png: 'image-link.png' }, coatOfArms: { png: 'image-link.png' }, id: '1',
    },
    {
      name: { common: 'country two' }, population: 1200, flags: { png: 'image-link.png' }, coatOfArms: { png: 'image-link.png' }, id: '2',
    },
    {
      name: { common: 'country three' }, population: 1534, flags: { png: 'image-link.png' }, coatOfArms: { png: 'image-link.png' }, id: '3',
    },
  ];

  beforeEach(() => {
    useSelector.mockReturnValue({ countries: mockData });
    useLocation.mockReturnValue({ state: '1' });
  });

  afterEach(() => {
    jest.clearAllMocks();
  });

  it('Should render correctly', () => {
    const tree = renderer.create(
      <Details />,
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('Should filter the selected country', () => {
    const location = useLocation();
    const filtered = mockData.filter((country) => country.id === location.state);
    expect(filtered.length).toBe(1);
  });
});
