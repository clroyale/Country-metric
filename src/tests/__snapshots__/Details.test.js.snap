// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Details Should render correctly 1`] = `
<div
  className="main_container"
>
  <div
    className="coat_of_arms_name_container"
  >
    <div
      className="coat_of_arms_container"
    >
      <img
        alt=""
        className="coat_of_arms"
        src="image-link.png"
      />
    </div>
    <div
      className="country_name_container"
    >
      <h2
        className="country_name"
      >
        country one
      </h2>
    </div>
  </div>
  <div>
    <h2
      className="break_down_text"
    >
      Information break down
    </h2>
  </div>
  <ul
    className="information_section"
  >
    <li
      className="info_list"
    >
      <span
        className="info_title"
      >
        Official name:
      </span>
    </li>
    <li
      className="info_list"
    >
      <span
        className="info_title"
      >
        Capital:
      </span>
    </li>
    <li
      className="info_list"
    >
      <span
        className="info_title"
      >
        Continent:
      </span>
    </li>
    <li
      className="info_list"
    >
      <span
        className="info_title"
      >
        Subregion:
      </span>
    </li>
    <li
      className="info_list"
    >
      <span
        className="info_title"
      >
        Population:
      </span>
      1234
    </li>
    <li
      className="info_list"
    >
      <span
        className="info_title"
      >
        Area:
      </span>
      undefined kmÂ²
    </li>
  </ul>
</div>
`;
